bokeh/events.py

```diff
@@ -0,0 +1,178 @@
+function with or without a Bokeh server. This can be accomplished with the
+:func:`~bokeh.model.Model.js_on_event` method:
+
+.. code-block:: python
+
+    # example setting up JS callback
+
+Alternatively it is possible to trigger Python code to run when events
+happen, in the context of a Bokeh application running on a Bokeh server.
+This can ccomplished with the :func:`~bokeh.model.Model.on_event` method:
+
+. code-block:: python
+
+    # example setting up python callback
+
+'''
+from __future__ import absolute_import
+
+class Event(object):
+    ''' Base class for all Bokeh events.
+
+    '''
+    event_classes = []
+    event_name = None
+
+    def __init__(self, model_id=None):
+        self.model_id = model_id
+
+    @classmethod
+    def register_event_class(cls, event_cls):
+        ''' Register a custom event class.
+
+        '''
+        cls.event_classes.append(event_cls)
+
+    @classmethod
+    def from_JSON(cls, json):
+        eventclasses = [eventcls for eventcls in cls.event_classes]
+        for eventscls in eventclasses:
+            if eventscls.event_name == json['event_name']:
+                return eventscls(**json['event_values'])
+        print('Warning: Could not find appropriate Event class')
+
+
+class ButtonClick(Event):
+    ''' Announce a button click event.
+
+    '''
+    event_name = 'button_click'
+
+    def __init__(self, model_id=None):
+        super(ButtonClick, self).__init__(model_id=model_id)
+
+
+class LODStart(Event):
+    ''' Announce the start of "interactive level-of-detail" mode on a plot.
+
+    '''
+    event_name = 'lodstart'
+
+class LODEnd(Event):
+    ''' Announce the end of "interactive level-of-detail" mode on a plot.
+
+    '''
+    event_name = 'lodend'
+
+class PointEvent(Event):
+    ''' Base class for UI events associated with a specific (x,y) point.
+
+    '''
+    event_name = None
+
+    def __init__(self, sx=None,sy=None, x=None, y=None, model_id=None):
+        self.sx = sx
+        self.sy = sy
+        self.x = x
+        self.y = y
+        super(PointEvent, self).__init__(model_id=model_id)
+
+# --- Point Events ------------------------------------------------------------
+
+class DoubleTap(PointEvent):
+    ''' Announce a double-tap or double-click event on a Bokeh plot.
+
+    '''
+    event_name = 'doubletap'
+
+class MouseEnter(PointEvent):
+    '''
+
+    '''
+    event_name = 'mouseenter'
+
+class MouseLeave(PointEvent):
+    '''
+
+    '''
+    event_name = 'mouseleave'
+
+class MouseMove(PointEvent):
+    '''
+
+    '''
+    event_name = 'mousemove'
+
+class MouseWheel(PointEvent):
+    '''
+
+    '''
+    event_name = 'wheel'
+
+    def __init__(self, delta=None, **kwargs):
+        self.delta = delta
+        super(MouseWheel, self).__init__(**kwargs)
+
+class Pan(PointEvent):
+    '''
+
+    '''
+    event_name = 'pan'
+
+    def __init__(self, deltaX=None, deltaY=None, direction=None, **kwargs):
+        self.deltaX = deltaX
+        self.deltaY = deltaY
```

Comment:
Here would you prefer ``deltax`` or ``delta_x``? Personally, I would lean towards the former...